x-pack/plugins/security_solution/server/lib/telemetry/sender.ts

```diff
@@ -383,6 +397,14 @@ export function copyAllowlistedFields(
     if (eventValue !== null && eventValue !== undefined) {
       if (allowValue === true) {
         return { ...newEvent, [allowKey]: eventValue };
+      } else if (typeof allowValue === 'object' && Array.isArray(eventValue)) {
+        return {
+          ...newEvent,
+          // cast to object[] to be a valid SearchTypes variant
+          [allowKey]: (eventValue as object[]).map((v) =>
+            copyAllowlistedFields(allowValue, v as TelemetryEvent)
+          ),
+        };
```

Comment:
```suggestion
        const allowedFields = [];
        for (const v of eventValue) {
          if (typeof v === 'object') {
            // casting this as AllowlistFields is a recursive structure, any object values it has are also AllowlistFields
            const sub: AllowlistFields = v as AllowlistFields;
            allowedFields.push(copyAllowlistedFields(allowValue, sub));
          }
        }
        return {
          ...newEvent,
          [allowKey]: allowedFields,
        };
```

This approach would check that the value is an object before casting it to an `AllowlistFields`

Reply:
thoughts on 64c9b7a9366ed8dc0337b763e9b576498f3918aa?

Reply:
also i thought for loops were basically despised as being non-idiomatic

Reply:
this work? basically did what you have but filter+map
bbcaec11a2d607853d861a2180a28b4c976c7245